ui/apps/platform/src/Containers/Violations/Details/Deployment/ContainerConfiguration.tsx

```diff
@@ -1,92 +1,38 @@
 import React, { ReactElement } from 'react';
-import { DescriptionList, Flex, FlexItem } from '@patternfly/react-core';
+import { Card, CardBody, CardTitle, Title } from '@patternfly/react-core';
 
-import DescriptionListItem from 'Components/DescriptionListItem';
-import { Container } from 'types/deployment.proto';
+import { Deployment } from 'types/deployment.proto';
+import ContainerConfigurationDescriptionList from './ContainerConfigurationDescriptionList';
 
-import ContainerImage from './ContainerImage';
-import ContainerResourcesDescriptionList from './ContainerResourcesDescriptionList';
-import ContainerSecretDescriptionList from './ContainerSecretDescriptionList';
-import ContainerVolumeDescriptionList from './ContainerVolumeDescriptionList';
+export type ContainerConfigurationProps = {
+    deployment: Deployment | null;
+};
 
-function MultilineDescription({ descArr }) {
-    return (
-        <Flex direction={{ default: 'column' }}>
-            {descArr.map((desc, idx) => (
-                // eslint-disable-next-line react/no-array-index-key
-                <div key={idx}>
-                    <FlexItem>{desc}</FlexItem>
-                </div>
-            ))}
-        </Flex>
-    );
-}
+function ContainerConfiguration({ deployment }: ContainerConfigurationProps): ReactElement {
+    let content: JSX.Element[] | string = 'None';
 
-type ContainerConfigurationProps = {
-    container: Container;
-};
+    if (deployment === null) {
+        content =
+            "Container configurations are unavailable because the alert's deployment no longer exists.";
+    } else {
+        if (Array.isArray(deployment?.containers) && deployment.containers.length !== 0) {
```

Comment:
Two nits:

1. The optional chaining should no longer be necessary since `deployment` is already checked to be non-null
2. Is the `Array.isArray` check needed here? Can `containers` be a non-array value, or was this only like this originally due to the previous need for optional chaining?

Reply:
I saw quite a bit of optional chaining I felt uneasy removing them 😅. But I think you're right that we can remove it for (1). I'm not really sure why there is an `Array.isArray`. I would like to assume it was due to some issue so I thought it was ok to leave it 🤔.